felix/bpf-gpl/xdp.c

```diff
@@ -61,6 +61,7 @@ int calico_xdp_main(struct xdp_md *xdp)
 		return XDP_DROP;
 	}
 	__builtin_memset(ctx->state, 0, sizeof(*ctx->state));
+	ctx->scratch = (void *)(ctx->xdp_globals + 1);
```

Comment:
What's going on here?

Reply:
although not used in XDP we need to have the pointer set, so pointing behind the globals as there is enough space.